server/src/main/java/io/crate/replication/logical/action/TransportAlterPublicationAction.java

```diff
@@ -164,6 +167,34 @@ static Publication updatePublication(Request request, Metadata currentMetadata,
         return new Publication(oldPublication.owner(), oldPublication.isForAllTables(), new ArrayList<>(tables));
     }
 
+    private static void updateSettings(IndexNameExpressionResolver indexNameExpressionResolver,
+                                       String publicationName,
+                                       Publication oldPublication,
+                                       Publication newPublication,
+                                       ClusterState currentState,
+                                       Metadata.Builder mdBuilder) {
+        // Check for removed tables
+        var oldTables = new ArrayList<>(oldPublication.tables());
+        oldTables.removeAll(newPublication.tables());
+        if (!oldTables.isEmpty()) {
+            applyPublicationSetting(indexNameExpressionResolver,
+                                    oldTables,
+                                    currentState,
+                                    mdBuilder,
+                                    x -> x.remove(publicationName));
+        }
+        // Check for new tables
+        var newTables = new ArrayList<>(newPublication.tables());
```

Comment:
After performing check for removed tables above, `oldTables` is mutated and can have side affect to tis new tables check. I think `newTables.removeAll(oldPublication.tables())` will always return false